genestack_client/scripts/genestack_application_manager.py

```diff
@@ -479,7 +479,11 @@ def upload_single_file(application, file_path, version, override,
         if result:
             print result
 
-        if release:
+        if release and stable and SCOPE_DICT[scope] == 'SYSTEM':
```

Comment:
In case then we have scope user and `-r` we will have next issue:

Application will be released with new version, and old version will be marked stable for user.